agent/structs/config_entry.go

```diff
@@ -335,6 +341,15 @@ func MakeConfigEntry(kind, name string) (ConfigEntry, error) {
 	}
 }
 
+func ValidateConfigEntryKind(kind string) bool {
```

Comment:
This should probably also be called in Validate in the agent config - it could return the error that's currently in the HTTP endpoint directly instead of a bool so that that doesn't have to be repeated.

Reply:
DecodeConfigEntry uses makeConfigEntry which already does the Kind validation.

This func is mainly needed in the case where we are listing entries and need to differentiate between simply having none created yet, and an invalid type that none can possibly be created for.

Reply:
We discussed this and the config builder is already doing similar validation.